azkaban-common/src/main/java/azkaban/project/ProjectManager.java

```diff
@@ -187,7 +171,8 @@ public Props getProps() {
       logger.error("Bad regex pattern " + regexPattern);
       return allProjects;
     }
-    for (Project project : getProjects()) {
+    List<Project> projects = getProjects();
+    for (Project project : projects) {
```

Comment:
why do we need to introduce projects here? why not just for (Project project : getProjects())